File: src/main/java/com/fish/entity/pojo/Account.java
package com.fish.entity.pojo;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.KeyType;
import com.mybatisflex.annotation.Table;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * @author fish
 * @since 2023-11-14
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(value = "oj_account")
public class Account implements Serializable {
    /**
     * 用户id
     */
    @Id(keyType = KeyType.Auto)
    private String userId;
    /**
     * 身份id
     */
    private Long roleId;
    /**
     * 昵称
     */
    private String nickname;
    /**
     * 头像
     */
    private String avatar;
    /**
     * 学号
     */
    private String studentId;
    /**
     * 姓名
     */
    private String username;
    /**
     * 密码
     */
    private String password;
    /**
     * 性别
     */
    private String gender;
    /**
     * 邮箱
     */
    private String email;
    /**
     * 专业
     */
    private String specialty;
    /**
     * QQ号
     */
    private String qq;
    /**
     * 博客地址
     */
    private String blogAddress;
    /**
     * GitHub地址
     */
    private String githubAddress;
    /**
     * 是否启用
     */
    private Boolean enabled;
    /**
     * 创建时间
     */
    @JsonSerialize(using = LocalDateTimeSerializer.class)
    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    private LocalDateTime createTime;
}

File: src/main/kotlin/com/fish/entity/pojo/Account.kt
package com.fish.entity.pojo

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import com.fasterxml.jackson.databind.annotation.JsonSerialize
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer
import com.fasterxml.jackson.datatype.jsr310.ser.LocalDateTimeSerializer
import com.mybatisflex.annotation.Id
import com.mybatisflex.annotation.KeyType
import com.mybatisflex.annotation.Table
import java.io.Serializable
import java.time.LocalDateTime

/**
 * @author fish
 * @since 2023-11-14
 */
@Table(value = "oj_account")
open class Account(
    userId: String?,
    roleId: Long?,
    nickname: String?,
    avatar: String?,
    studentId: String?,
    username: String?,
    password: String?,
    gender: String?,
    email: String?,
    specialty: String?,
    qq: String?,
    blogAddress: String?,
    githubAddress: String?,
    enabled: Boolean?,
    createTime: LocalDateTime?
): Serializable {
    /**
     * 用户id
     */
    @Id(keyType = KeyType.Auto)
    val userId: String? = null
    /**
     * 身份id
     */
    val roleId: Long? = null
    /**
     * 昵称
     */
    val nickname: String? = null
    /**
     * 头像
     */
    val avatar: String? = null
    /**
     * 学号
     */
    val studentId: String? = null
    /**
     * 姓名
     */
    val username: String? = null
    /**
     * 密码
     */
    val password: String? = null
    /**
     * 性别
     */
    val gender: String? = null
    /**
     * 邮箱
     */
    val email: String? = null
    /**
     * 专业
     */
    val specialty: String? = null
    /**
     * QQ号
     */
    val qq: String? = null
    /**
     * 博客地址
     */
    val blogAddress: String? = null
    /**
     * GitHub地址
     */
    val githubAddress: String? = null
    /**
     * 是否启用
     */
    val enabled: Boolean? = null
    /**
     * 创建时间
     */
    @JsonSerialize(using = LocalDateTimeSerializer::class)
    @JsonDeserialize(using = LocalDateTimeDeserializer::class)
    val createTime: LocalDateTime? = null
}
